Sources/SwiftProtobuf/Google_Protobuf_Timestamp_Extensions.swift

```diff
@@ -270,49 +266,53 @@ public extension Google_Protobuf_Timestamp {
     public init(decoder: inout JSONDecoder) throws {
         self.init()
         let s = try decoder.scanner.nextQuotedString()
-        let timestamp = try parseTimestamp(s: s)
-        seconds = timestamp.0
-        nanos = timestamp.1
+        (seconds, nanos) = try parseTimestamp(s: s)
     }
 
     public func serializeJSON() throws -> String {
-        let s = seconds
-        let n = nanos
-        if let formatted = formatTimestamp(seconds: s, nanos: n) {
+        if let formatted = formatTimestamp(seconds: seconds, nanos: nanos) {
             return "\"\(formatted)\""
         } else {
             throw EncodingError.timestampJSONRange
         }
     }
 
-    func serializeAnyJSON() throws -> String {
+    public func serializeAnyJSON() throws -> String {
         let value = try serializeJSON()
         return "{\"@type\":\"\(anyTypeURL)\",\"value\":\(value)}"
     }
 }
 
-private func normalizedTimestamp(seconds: Int64, nanos: Int32) -> Google_Protobuf_Timestamp {
-    var s = seconds
-    var n = nanos
-    if n >= 1000000000 || n <= -1000000000 {
-        s += Int64(n) / 1000000000
-        n = n % 1000000000
+private func normalizeForTimestamp(seconds: Int64, nanos: Int32) -> (seconds: Int64, nanos: Int32) {
+    // The Timestamp spec says that nanos must be in the range [0, 999999999), as in actual
+    // modular arithmetic. Integer % and / do not reflect this, so mod() and div() are defined
+    // for this purpose. Factored out, they make normalization very straightforward.
```

Comment:
Well, right, that works, but I already have a follow-up change to factor out some general purpose math and time functions (including div and mod), at which point this code becomes:

s = s + div(n,x)
n = mod(n,x)

which seems simpler to me. In the sense that it expresses the actual definition of the thing being computed. Do you agree? Then I get to reuse those functions to clear up other inscrutable arithmetic code around here (look at the computation of secondsSinceStartOfDay for instance). I can pull those changes into this PR if you like.